src/Repositories/ExtensionCacheRepository.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Flagrow\Bazaar\Repositories;
+
+use Flagrow\Bazaar\Extensions\ExtensionUtils;
+use Flagrow\Bazaar\Search\FlagrowApi;
+use Flagrow\Bazaar\Traits\Cachable;
+use Illuminate\Contracts\Cache\Store;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Psr\Http\Message\ResponseInterface;
+
+class ExtensionCacheRepository
+{
+    use Cachable;
+
+    /**
+     * @var Store
+     */
+    protected $store;
+
+    /**
+     * @var FlagrowApi
+     */
+    protected $api;
+
+    const CACHE_KEY = 'flagrow.io.packages';
+    const CACHE_EXPIRE = 60; // minutes
+
+    public function __construct(Store $store, FlagrowApi $api)
+    {
+        $this->store = $store;
+        $this->api = $api;
+    }
+
+    /**
+     * Extracts the data from an API response
+     * @param ResponseInterface $response
+     * @return array
+     */
+    protected function unpackResponseData(ResponseInterface $response)
+    {
+        $json = json_decode($response->getBody()->getContents(), true);
+
+        return Arr::get($json, 'data', []);
+    }
+
+    /**
+     * (re)freshes extension data from the api
+     * @return array
+     */
+    protected function allExtensionsFromApi()
```

Comment:
I don't think a Cache related class should have any other responsibility. Building the query should probably now be a responsibility of the Flagrow API client.

Reply:
In fact... I'm not sure it's worth it. Most of the `allExtensionsFromApi` method has nothing to do with the API and all to do with caching.

The only things that would change is `$this->api->get('packages', compact('query'))` that would become something like `$this->api->listPackages($query)`. Is there any advantage of doing that ?

By the way, the `FlagrowApi` class isn't really an API wrapper, it's just an HTTP client configured for the Flagrow Api :see_no_evil:

Reply:
At some point we'll introduce search more heavily. Having all that logic inside a cache repository will be confusing, especially if you introduce this confusion during a rewrite meant to lower it. I hope you see the sense of that.
I honestly don't mind to move it to the api (wrapper/client) (temporarily). At some point it will hold all logic, we could create methods for our calls in there instead of spreading them out over the handlers/repositories. Favourite and the review functionality for instance.

Please note if you don't take it in now, it will most likely be done in the future anyway. I just think and stick to the fact that a cache repository has no responsibility at all over other logic.

😁